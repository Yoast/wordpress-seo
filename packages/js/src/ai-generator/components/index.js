export { App } from "./app";
export { GoogleContent } from "./google-content";
export { GooglePreview } from "./google-preview";
export { GooglePreviewSkeleton } from "./google-preview-skeleton";
export { Introduction } from "./introduction";
export { LengthProgressBar } from "./length-progress-bar";
export { ModalContent } from "./modal-content";
export { PreviewModePicker } from "./preview-mode-picker";
export { SocialContent } from "./social-content";
export { SocialPreview } from "./social-preview";
export { SocialPreviewSkeleton } from "./social-preview-skeleton";
export { SuggestionError } from "./suggestion-error";
export { SuggestionsList } from "./suggestions-list";
export { SuggestionsListSkeleton } from "./suggestions-list-skeleton";
export { TipNotification } from "./tip-notification";
export { TwitterContent } from "./twitter-content";
export { TwitterPreview } from "./twitter-preview";
export { TwitterPreviewSkeleton } from "./twitter-preview-skeleton";
export { TypeProvider } from "./type-provider";
export { SparksLimitNotification } from "./sparks-limit-notification";
