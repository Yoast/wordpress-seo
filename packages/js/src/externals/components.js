import CollapsibleCornerstone from "../components/CollapsibleCornerstone";
import KeywordInput from "../components/contentAnalysis/KeywordInput";
import ReadabilityAnalysis from "../components/contentAnalysis/ReadabilityAnalysis";
import SeoAnalysis from "../components/contentAnalysis/SeoAnalysis";

window.yoast = window.yoast || {};
window.yoast.externals = window.yoast.externals || {};
window.yoast.externals.components = {
	CollapsibleCornerstone,
	KeywordInput,
	ReadabilityAnalysis,
	SeoAnalysis,
};
