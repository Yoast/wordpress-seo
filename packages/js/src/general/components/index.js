export { AlertsList } from "./alerts-list";
export { AlertsTitle } from "./alerts-title";
export { Collapsible } from "./collapsible";
export { Notice } from "./notice";
export { Notifications } from "./notifications";
export { Problems } from "./problems";
export { RouteErrorFallback } from "./route-error-fallback";
export { RouteLayout } from "./route-layout";
