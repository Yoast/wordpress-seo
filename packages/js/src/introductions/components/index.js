export { Content } from "./content";
export { Introduction } from "./introduction";
export { Modal } from "./modal";
export { IntroductionProvider, useIntroductionsContext } from "./provider";
