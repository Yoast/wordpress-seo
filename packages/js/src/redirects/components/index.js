export { RouteErrorFallback } from "./route-error-fallback";
export { RouteLayout } from "./route-layout";
export { Menu } from "./menu";
export { SidebarLayout } from "./sidebar-layout";
export { Notifications } from "./notifications";
