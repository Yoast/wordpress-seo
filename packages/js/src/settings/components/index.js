export { default as FieldsetLayout } from "./fieldset-layout";
export { default as FormLayout } from "./form-layout";
export { default as FormikFlippedToggleField } from "./formik-flipped-toggle-field";
export { default as FormikMediaSelectField } from "./formik-media-select-field";
export { default as FormikReplacementVariableEditorField } from "./formik-replacement-variable-editor-field";
export { default as FormikValueChangeField } from "./formik-value-change-field";
export { default as FormikWithErrorField } from "./formik-with-error-field";
export { default as Notifications } from "./notifications";
export { default as OpenGraphDisabledAlert } from "./open-graph-disabled-alert";
export { default as SidebarNavigation } from "./sidebar-navigation";
export { ReactComponent as YoastLogo } from "./yoast-logo.svg";
