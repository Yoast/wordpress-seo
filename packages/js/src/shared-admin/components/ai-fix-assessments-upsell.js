import { LockOpenIcon } from "@heroicons/react/outline";
import { ArrowNarrowRightIcon } from "@heroicons/react/solid";
import { createInterpolateElement } from "@wordpress/element";
import { __, sprintf } from "@wordpress/i18n";
import { Badge, Button, useModalContext } from "@yoast/ui-library";
import PropTypes from "prop-types";
import { OutboundLink, VideoFlow } from ".";

/**
 * @param {string} learnMoreLink The learn more link.
 * @param {Object} thumbnail The thumbnail: img props.
 * @param {Object} wistiaEmbedPermission The value, status and set for the Wistia embed permission.
 * @param {string} upsellLink The upsell link.
 * @param {string} upsellLabel The upsell label.
 * @returns {JSX.Element} The element.
 */
export const AiFixAssessmentsUpsell = ( {
	learnMoreLink,
	thumbnail,
	wistiaEmbedPermission,
	upsellLink,
	upsellLabel,
} ) => {
	const { onClose, initialFocus } = useModalContext();

	const learnMoreLinkStructure = {
		a: <OutboundLink
			href={ learnMoreLink }
			className="yst-inline-flex yst-items-center yst-gap-1 yst-no-underline yst-font-medium"
			variant="primary"
		/>,
		ArrowNarrowRightIcon: <ArrowNarrowRightIcon className="yst-w-4 yst-h-4 rtl:yst-rotate-180" />,
		br: <br />,
	};

	return (
		<>
			<div className="yst-px-10 yst-pt-10 yst-introduction-gradient yst-text-center">
				<div className="yst-relative yst-w-full">
					<VideoFlow
						videoId="vun9z1dpfh"
						thumbnail={ thumbnail }
						wistiaEmbedPermission={ wistiaEmbedPermission }
					/>
					<Badge className="yst-absolute yst-end-4 yst-text-center yst-justify-center" variant="info" style={ { top: "-8px" } }>
						{ __( "Beta", "wordpress-seo-premium" ) }
					</Badge>
				</div>
				<div className="yst-mt-6 yst-text-xs yst-font-medium yst-flex yst-flex-col yst-items-center">
					<span className="yst-introduction-modal-uppercase yst-flex yst-gap-2 yst-items-center">
						<span className="yst-logo-icon" />
						Yoast SEO Premium
					</span>
				</div>
			</div>
			<div className="yst-px-10 yst-pb-4 yst-flex yst-flex-col yst-items-center">
				<div className="yst-mt-4 yst-mx-1.5 yst-text-center">
					<h3 className="yst-text-slate-900 yst-text-lg yst-font-medium">
						{
							sprintf(
								/* translators: %s: Expands to "Yoast AI" */
								__( "Optimize your SEO content with %s", "wordpress-seo" ),
								"Yoast AI"
							)
						}
					</h3>
					<div className="yst-mt-2 yst-text-slate-600 yst-text-sm">
						{ createInterpolateElement(
							sprintf(
								/* translators: %1$s is a break tag; %2$s and %3$s are anchor tags; %4$s is the arrow icon. */
								__(
									"Make content editing a breeze! Optimize your SEO content with quick, actionable suggestions at the click of a button.%1$s%2$sLearn more%3$s%4$s",
									"wordpress-seo"
								),
								"<br/>",
								"<a>",
								"<ArrowNarrowRightIcon />",
								"</a>"
							),
							learnMoreLinkStructure
						) }
					</div>
				</div>
				<div className="yst-w-full yst-flex yst-mt-6">
					<Button
						as="a"
						className="yst-grow"
						size="extra-large"
						variant="upsell"
						href={ upsellLink }
						target="_blank"
						ref={ initialFocus }
					>
						<LockOpenIcon className="yst--ms-1 yst-me-2 yst-h-5 yst-w-5" />
						{ upsellLabel }
						<span className="yst-sr-only">
							{
								/* translators: Hidden accessibility text. */
								__( "(Opens in a new browser tab)", "wordpress-seo" )
							}
						</span>
					</Button>
				</div>
				<Button
					as="a"
					className="yst-mt-4"
					variant="tertiary"
					onClick={ onClose }
				>
					{ __( "Close", "wordpress-seo" ) }
				</Button>
			</div>
		</>
	);
};
AiFixAssessmentsUpsell.propTypes = {
	learnMoreLink: PropTypes.string.isRequired,
	upsellLink: PropTypes.string.isRequired,
	thumbnail: PropTypes.shape( {
		src: PropTypes.string.isRequired,
		width: PropTypes.string,
		height: PropTypes.string,
	} ).isRequired,
	wistiaEmbedPermission: PropTypes.shape( {
		value: PropTypes.bool.isRequired,
		status: PropTypes.string.isRequired,
		set: PropTypes.func.isRequired,
	} ).isRequired,
	upsellLabel: PropTypes.string,
};

AiFixAssessmentsUpsell.defaultProps = {
	upsellLabel: sprintf(
		/* translators: %1$s expands to Yoast SEO Premium. */
		__( "Unlock with %1$s", "wordpress-seo" ),
		"Yoast SEO Premium"
	),
};
