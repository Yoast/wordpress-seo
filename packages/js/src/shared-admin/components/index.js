export { AcademyUpsellCard } from "./academy-upsell-card";
export { AiGenerateTitlesAndDescriptionsUpsell } from "./ai-generate-titles-and-descriptions-upsell";
export { AiFixAssessmentsUpsell } from "./ai-fix-assessments-upsell";
export { ErrorFallback } from "./error-fallback";
export { FieldsetLayout } from "./fieldset-layout";
export { ReactComponent as G2LogoWhite } from "./g2-logo-white.svg";
export { MenuItemLink } from "./menu-item-link";
export { OutboundLink } from "./outbound-link";
export { PremiumUpsellCard } from "./premium-upsell-card";
export { PremiumUpsellList } from "./premium-upsell-list";
export { SidebarRecommendations } from "./sidebar-recommendations";
export { UnsavedChangesModal } from "./unsaved-changes-modal";
export { VideoFlow } from "./video-flow";
export { ReactComponent as YoastLogo } from "./yoast-logo.svg";
export { SiteKitConsentModal } from "./site-kit-consent-modal";
