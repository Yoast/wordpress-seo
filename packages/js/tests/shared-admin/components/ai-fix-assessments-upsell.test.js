import { render } from "../../test-utils";
import { AiFixAssessmentsUpsell } from "../../../src/shared-admin/components/ai-fix-assessments-upsell";

describe( "AiFixAssessmentsUpsell", () => {
	const props = {
		learnMoreLink: "https://example.com/learn-more",
		thumbnail: {
			src: "thumbnail.jpg",
			width: "100",
			height: "100",
		},
		wistiaEmbedPermission: {
			value: true,
			status: "granted",
			set: jest.fn(),
		},
		upsellLink: "https://example.com/upsell",
		upsellLabel: "Test Upsell Label",
	};

	it( "renders the component correctly", () => {
		const { container } = render( <AiFixAssessmentsUpsell { ...props } /> );
		expect( container ).toMatchSnapshot();
	} );
} );
