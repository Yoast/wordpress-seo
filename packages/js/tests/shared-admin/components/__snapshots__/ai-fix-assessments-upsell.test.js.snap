// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AiFixAssessmentsUpsell renders the component correctly 1`] = `
<div>
  <div
    class="yst-px-10 yst-pt-10 yst-introduction-gradient yst-text-center"
  >
    <div
      class="yst-relative yst-w-full"
    >
      <div
        class="yst-relative yst-w-full yst-h-0 yst-pt-[56.25%] yst-overflow-hidden yst-rounded-md yst-drop-shadow-md yst-bg-white"
      >
        <div
          class="yst-absolute yst-w-full yst-h-full yst-top-0 yst-right-0"
        >
          <div
            class="wistia_embed wistia_async_vun9z1dpfh videoFoam=true"
          />
        </div>
      </div>
      <span
        class="yst-badge yst-badge--info yst-absolute yst-end-4 yst-text-center yst-justify-center"
        style="top: -8px;"
      >
        Beta
      </span>
    </div>
    <div
      class="yst-mt-6 yst-text-xs yst-font-medium yst-flex yst-flex-col yst-items-center"
    >
      <span
        class="yst-introduction-modal-uppercase yst-flex yst-gap-2 yst-items-center"
      >
        <span
          class="yst-logo-icon"
        />
        Yoast SEO Premium
      </span>
    </div>
  </div>
  <div
    class="yst-px-10 yst-pb-4 yst-flex yst-flex-col yst-items-center"
  >
    <div
      class="yst-mt-4 yst-mx-1.5 yst-text-center"
    >
      <h3
        class="yst-text-slate-900 yst-text-lg yst-font-medium"
      >
        Optimize your SEO content with Yoast AI
      </h3>
      <div
        class="yst-mt-2 yst-text-slate-600 yst-text-sm"
      >
        Make content editing a breeze! Optimize your SEO content with quick, actionable suggestions at the click of a button.
        <br />
        <a
          class="yst-link yst-link--primary yst-inline-flex yst-items-center yst-gap-1 yst-no-underline yst-font-medium"
          href="https://example.com/learn-more"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn more
          <svg
            aria-hidden="true"
            class="yst-w-4 yst-h-4 rtl:yst-rotate-180"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12.293 5.293a1 1 0 011.414 0l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-2.293-2.293a1 1 0 010-1.414z"
              fill-rule="evenodd"
            />
          </svg>
          <span
            class="yst-sr-only"
          >
            (Opens in a new browser tab)
          </span>
        </a>
      </div>
    </div>
    <div
      class="yst-w-full yst-flex yst-mt-6"
    >
      <a
        class="yst-button yst-button--upsell yst-button--extra-large yst-grow"
        href="https://example.com/upsell"
        target="_blank"
      >
        <svg
          aria-hidden="true"
          class="yst--ms-1 yst-me-2 yst-h-5 yst-w-5"
          fill="none"
          stroke="currentColor"
          stroke-width="2"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 11V7a4 4 0 118 0m-4 8v2m-6 4h12a2 2 0 002-2v-6a2 2 0 00-2-2H6a2 2 0 00-2 2v6a2 2 0 002 2z"
            stroke-linecap="round"
            stroke-linejoin="round"
          />
        </svg>
        Test Upsell Label
        <span
          class="yst-sr-only"
        >
          (Opens in a new browser tab)
        </span>
      </a>
    </div>
    <a
      class="yst-button yst-button--tertiary yst-mt-4"
    >
      Close
    </a>
  </div>
</div>
`;
